app/command_autocomplete.go

```diff
@@ -0,0 +1,253 @@
+	namedArg := "--" + arg.Name
+	if in == "" { //The user has not started typing the argument.
+		return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed + namedArg + " ", Suggestion: namedArg, Hint: "", Description: arg.HelpText}
+	}
+	if strings.HasPrefix(strings.ToLower(namedArg), strings.ToLower(in)) {
+		return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed + namedArg[len(in):] + " ", Suggestion: namedArg, Hint: "", Description: arg.HelpText}
+	}
+
+	if !strings.HasPrefix(strings.ToLower(in), strings.ToLower(namedArg)+" ") {
+		return false, parsed + toBeParsed, "", model.AutocompleteSuggestion{}
+	}
+	if strings.ToLower(in) == strings.ToLower(namedArg)+" " {
+		return false, parsed + namedArg + " ", " ", model.AutocompleteSuggestion{}
+	}
+	return false, parsed + namedArg + " ", in[len(namedArg)+1:], model.AutocompleteSuggestion{}
+}
+
+func parseInputTextArgument(arg *model.AutocompleteArg, parsed, toBeParsed string) (found bool, alreadyParsed string, yetToBeParsed string, suggestion model.AutocompleteSuggestion) {
+	in := strings.TrimPrefix(toBeParsed, " ")
+	a := arg.Data.(*model.AutocompleteTextArg)
+	if in == "" { //The user has not started typing the argument.
+		return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed, Suggestion: "", Hint: a.Hint, Description: arg.HelpText}
+	}
+	if in[0] == '"' { //input with multiple words
+		indexOfSecondQuote := strings.Index(in[1:], `"`)
+		if indexOfSecondQuote == -1 { //typing of the multiple word argument is not finished
+			return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed, Suggestion: "", Hint: a.Hint, Description: arg.HelpText}
+		}
+		// this argument is typed already
+		offset := 2
+		if len(in) > indexOfSecondQuote+2 && in[indexOfSecondQuote+2] == ' ' {
+			offset++
+		}
+		return false, parsed + in[:indexOfSecondQuote+offset], in[indexOfSecondQuote+offset:], model.AutocompleteSuggestion{}
+	}
+	// input with a single word
+	index := strings.Index(in, " ")
+	if index == -1 { // typing of the single word argument is not finished
+		return true, parsed + toBeParsed, "", model.AutocompleteSuggestion{Complete: parsed + toBeParsed, Suggestion: "", Hint: a.Hint, Description: arg.HelpText}
+	}
+	// single word argument already typed
+	return false, parsed + in[:index+1], in[index+1:], model.AutocompleteSuggestion{}
+}
+
+func parseStaticListArgument(arg *model.AutocompleteArg, parsed, toBeParsed string) (found bool, alreadyParsed string, yetToBeParsed string, suggestions []model.AutocompleteSuggestion) {
+	a := arg.Data.(*model.AutocompleteStaticListArg)
+	return parseListItems(a.PossibleArguments, parsed, toBeParsed)
+}
+
+func (a *App) parseDynamicListArgument(arg *model.AutocompleteArg, parsed, toBeParsed string) (found bool, alreadyParsed string, yetToBeParsed string, suggestions []model.AutocompleteSuggestion) {
```

Comment:
0/5 I doesn't seam like there is much parsing
```suggestion
func (a *App) getDynamicListArgument(arg *model.AutocompleteArg, parsed, toBeParsed string) (found bool, alreadyParsed string, yetToBeParsed string, suggestions []model.AutocompleteSuggestion) {
```